### PR TITLE
Split `TransactionViewBalance` component

### PR DESCRIPTION
**Pull request by Gudahtt**
The `TransactionViewBalance` component has been split into three separate components. This was done primarily to make the asset page easier to implement. Also the name `TransactionViewBalance` didn't describe this component very well anymore.

Instead of the Ethereum and token-specific logic being in the same component, the two cases were split into the `EthOverview` and `TokenOverview` components respectively. They both use the `WalletOverview` component, which has the structure shared by both cases.

### CODE REVIEW

**Comment by Gudahtt** [comment]
There should be no user-facing changes in this PR. I've compared this to develop (boht the popup view and fullscreen) and didn't notice any differences.

File: ui/app/components/app/wallet-overview/eth-overview.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        </Tooltip>
+      )}
+      buttons={(
+        <>
+          <Button
+            type="secondary"
+            className="eth-overview__button"
+            onClick={() => {
+              metricsEvent({
+                eventOpts: {
```
Comment (brad-decker) [inline ui/app/components/app/wallet-overview/eth-overview.js:71]: The `useMetricsEvent` api is a little different than just exposing a context function. You'd want to use that hook multiple times for each payload you want to use, it returns a memoized callback function that you can just pass to your click handler. 

eg:
https://github.com/MetaMask/metamask-extension/blob/8a0230f7437b8704e4289d534d1f5080e26c81c0/ui/app/hooks/useRetryTransaction.js#L32-L38

File: ui/app/components/app/wallet-overview/eth-overview.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+              dispatch(showModal({ name: 'DEPOSIT_ETHER' }))
+            }}
+          >
+            { t('deposit') }
+          </Button>
+          <Button
+            type="secondary"
+            className="eth-overview__button"
+            onClick={() => {
+              metricsEvent({
```
Comment (brad-decker) [inline ui/app/components/app/wallet-overview/eth-overview.js:86]: here too

File: ui/app/components/app/wallet-overview/token-overview.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+          />
+        </div>
+      )}
+      buttons={(
+        <Button
+          type="secondary"
+          className="token-overview__button"
+          onClick={() => {
+            metricsEvent({
+              eventOpts: {
```
Comment (brad-decker) [inline ui/app/components/app/wallet-overview/token-overview.js:38]: and here

**Review by brad-decker — CHANGES_REQUESTED** [review]

File: ui/app/components/app/wallet-overview/eth-overview.js (hunk 1 above)
Comment (Gudahtt) [reply ui/app/components/app/wallet-overview/eth-overview.js:71]: Whoops! I'll fix this.

Comment (Gudahtt) [reply ui/app/components/app/wallet-overview/eth-overview.js]: All three spots have been updated.

Comment (Gudahtt) [reply ui/app/components/app/wallet-overview/eth-overview.js]: I used this incorrectly in the AssetList component as well apparently. Fixed in #8639

**Review by brad-decker — APPROVED** [review]
LGTM

**Review by whymarrh — APPROVED** [review]
LGTM!

**Pull request merged**